src/FamilyHubs.ServiceDirectory.Data/Repository/ApplicationDbContext.cs

```diff
@@ -44,6 +45,105 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .IsClustered(false);
             });
 
+            modelBuilder.Entity<ServiceSearch>()
+                .ToTable("ServiceSearches")
+                .HasKey(e => e.Id);
+
+            modelBuilder.Entity<ServiceSearch>()
+                .Property(e => e.SearchTriggerEventId)
+                .HasConversion<short>();
+            
+            modelBuilder.Entity<ServiceSearch>()
+                .Property(e => e.ServiceSearchTypeId)
+                .HasConversion<byte>();
+            
+            modelBuilder.Entity<ServiceSearch>()
+                .HasOne(e => e.SearchTriggerEvent)
+                .WithMany(e => e.ServiceSearches)
+                .HasForeignKey(e => e.SearchTriggerEventId)
+                .IsRequired(false);
+            
+            modelBuilder.Entity<ServiceSearch>()
+                .HasOne(e => e.ServiceSearchType)
+                .WithMany(e => e.ServiceSearches)
+                .HasForeignKey(e => e.ServiceSearchTypeId)
+                .IsRequired(true);
+
+            modelBuilder.Entity<ServiceSearchResult>()
+                .ToTable("ServiceSearchResults")
+                .HasKey(e => e.Id);
+
+            modelBuilder.Entity<ServiceSearchResult>()
+                .HasOne(e => e.Service)
+                .WithMany(e => e.ServiceSearchResults)
+                .HasForeignKey(e => e.ServiceId)
+                // Do not delete metrics if service deleted
+                .OnDelete(DeleteBehavior.NoAction);
+
+            modelBuilder.Entity<Event>()
+                .ToTable("Events")
+                .HasKey(e => e.Id);
+
+            modelBuilder.Entity<Event>()
+                .Property(e => e.Id)
+                .HasConversion<short>();
+            
+            modelBuilder.Entity<Event>()
+                .Property(e => e.Name)
+                .HasColumnType("nvarchar(100)");
```

Comment:
On the back of Gary's comment that the SQL it generated is just doing `nvarchar(max)`, weird that EF Core isn't picking this up properly, but it might be worth trying something like:

`modelBuilder.Entity<Event>().Property(e => e.Name).HasMaxLength(100);`

Reply:
Nice, `HasMaxLength` seemed to work. Thanks!